src/polyfem/time_integrator/ImplicitTimeIntegrator.cpp

```diff
@@ -56,16 +56,25 @@ namespace polyfem
 			dt_ = dt;
 		}
 
-		void ImplicitTimeIntegrator::save_raw(const std::string &x_path, const std::string &v_path, const std::string &a_path) const
+		void ImplicitTimeIntegrator::save_raw(const std::string &state_path, const std::string &x_path, const std::string &v_path, const std::string &a_path) const
 		{
-			if (!x_path.empty())
-				write_matrix(x_path, x_prev());
+			if (!state_path.empty())
+			{
+				write_matrix(state_path, "u", x_prev());
```

Comment:
This should save all previous quantities that the time-integrator is using.

Reply:
yes...

Reply:
I fixed this by saving a matrix instead of a vector. There is one hacky part where I load the solution twice: first ignoring any previous values and then again keeping the values and passing them to the time integrator.